src/go/rpk/pkg/cli/topic/config.go

```diff
@@ -71,6 +71,34 @@ valid, but does not apply it.
 			subtractKVs, err := parseKVs(subtracts)
 			out.MaybeDie(err, "unable to parse --subtract: %v", err)
 
+			// Redpanda fails to set both remote.read and write when passed
+			// at the same time, so we issue first the set request for write,
+			// then the rest of the requests.
+			// See https://github.com/redpanda-data/redpanda/issues/9191
+			_, isRRR := setKVs["redpanda.remote.read"]
+			wv, isRRW := setKVs["redpanda.remote.write"]
+			if isRRR && isRRW {
+				req := kmsg.NewPtrIncrementalAlterConfigsRequest()
+				req.ValidateOnly = dry
+				cfg := kmsg.NewIncrementalAlterConfigsRequestResourceConfig()
+				cfg.Name = "redpanda.remote.write"
+				cfg.Op = kmsg.IncrementalAlterConfigOpSet
+				cfg.Value = kmsg.StringPtr(wv)
+
+				for _, topic := range topics {
+					reqTopic := kmsg.NewIncrementalAlterConfigsRequestResource()
+					reqTopic.ResourceType = kmsg.ConfigResourceTypeTopic
+					reqTopic.ResourceName = topic
+					reqTopic.Configs = []kmsg.IncrementalAlterConfigsRequestResourceConfig{cfg}
+					req.Resources = append(req.Resources, reqTopic)
+				}
+
+				_, err := req.RequestWith(context.Background(), cl)
+				out.MaybeDie(err, "unable to incrementally update configs: %v", err)
```

Comment:
This loses any resp.Resources[*].ErrorCodes. Instead what can happen: save a `rrwErrors := make(map[string]int16)`, then below when looping through topics printing them, if resource.ErrorCode != 0 || rrwErrors[topic] != 0, use a non-zero error. You could get fancier and save an error message as well,

```go
type rrwErr struct {
    code int16
    msg *string
}
rrwErrs := make(map[string]rrwErr)
```
then apply below as appropriate

Reply:
Good catch, I'll add that... but overall, +1 to sending 2 requests?

Reply:
Fixed, thanks.